iothub/device/src/IotHubBaseClient.cs

```diff
@@ -128,6 +128,11 @@ public abstract class IotHubBaseClient : IAsyncDisposable
         /// <exception cref="OperationCanceledException">The operation has been canceled.</exception>
         /// <exception cref="IotHubClientException">An error occured when communicating with IoT hub service.</exception>
         /// <exception cref="ObjectDisposedException">The client has been disposed.</exception>
+        /// <example>
+        /// <code language="csharp">
+        /// await client.OpenAsync(cancellationToken);
```

Comment:
Not normally, no. Our client will retry retryable errors for this API. There is no end retry count for our default policy. It will either continue with no end, or until a passed cancellation token is cancelled. So, a user shouldn't require anything additional to be done _normally_.

Reply:
Not if the cancellation token is signalled.

I think what bothers me about this example is that it is too naive and straight forward. If the device doesn't connect then the IoT device is out of commission and may need human intervention.

Without having a whole reconnection sample here (which is way overkill), there are 2 things I think we can do:

1. Put the call to open in context. The idea is we initialize the client, subscribe to connection status changes, and then call open. Nothing else can happen before those 3 steps, and setting the connection status callback is best practice so you are notified that you are connected. I'd suggest we add all 3 statements to this example and end with a comment to begin device operations, such as telemetry.

2. Add a remark that the call to open is important and may require some amount of retry above and beyond the client retry policy (especially if it is disabled or if the cancellation token is signaled). One might choose to use a generous cancellation token but then (if possible) restart networking and try again. Later, once we're in main, we can add a link to the reconnection sample.

Does that sound reasonable?